hector_gazebo_plugins/src/gazebo_ros_force_based_move.cpp

```diff
@@ -358,25 +360,31 @@ namespace gazebo
   }
 
 
-  tf::Transform GazeboRosForceBasedMove::getTransformForMotion(double linear_vel_x, double angular_vel, double timeSeconds) const
+  tf::Transform GazeboRosForceBasedMove::getTransformForMotion(double linear_vel_x, double linear_vel_y, double angular_vel, double timeSeconds) const
   {
     tf::Transform tmp;
     tmp.setIdentity();
 
 
     if (std::abs(angular_vel) < 0.0001) {
       //Drive straight
-      tmp.setOrigin(tf::Vector3(static_cast<double>(linear_vel_x*timeSeconds), 0.0, 0.0));
+      tmp.setOrigin(tf::Vector3(static_cast<double>(linear_vel_x*timeSeconds), static_cast<double>(linear_vel_y*timeSeconds), 0.0));
     } else {
       //Follow circular arc
-      double distChange = linear_vel_x * timeSeconds;
-      double angleChange = angular_vel * timeSeconds;
+      const double distX = linear_vel_x * timeSeconds;
+      const double distY = linear_vel_y * timeSeconds;
+      const double distChange = std::sqrt(distX * distX + distY * distY);
+      const double angleDist = std::acos(distX / distChange);
```

Comment:
The naming of this variable is somewhat unintuitive.
I would suggest renaming to something like `angleDriveDirection`.